NtupleProducer/plugins/HTauTauNtuplizer.cc

```diff
@@ -2942,6 +2968,89 @@ void HTauTauNtuplizer::FillFatJet(const edm::View<pat::Jet>* fatjets, const edm:
         }
       }
       _ak8jets_nsubjets.push_back(nsubj);
+
+    // PF jet ID for AK8 Puppi (same as AK4 Puppi)
+    float NHF                 = ijet->neutralHadronEnergyFraction();
+    float NEMF                = ijet->neutralEmEnergyFraction();
+    float CHF                 = ijet->chargedHadronEnergyFraction();
+    float MUF                 = ijet->muonEnergyFraction();
+    float CEMF                = ijet->chargedEmEnergyFraction();
+    int   NumNeutralParticles = ijet->neutralMultiplicity();
+    int   chargedMult         = ijet->chargedMultiplicity();
+    int   NumConst            = ijet->chargedMultiplicity()+NumNeutralParticles;
+    float CHM                 = ijet->chargedMultiplicity();
+    float absjeta             = fabs(ijet->eta());
```

Comment:
Looking quickly at the code I don't see any separation between AK8 and AK4 jets. ```ijet``` could refer to any?

Reply:
true, the jet was already called `ijet` in the fatjet loop before, but I can also change it if you'd like